x/gamm/pool-models/stableswap/pool_test.go

```diff
@@ -505,3 +505,137 @@ func TestScaleCoin(t *testing.T) {
 		})
 	}
 }
+
+func TestSwapOutAmtGivenIn(t *testing.T) {
```

Comment:
Note: these functions will be more comprehensively tested in #2346 – I only added cases relating to the changes made in this PR so it doesn't get blocked